LBHackney-IT/lbh-income-api
# Working day fix

### Context
Currently when previewing letters*, the date on the letters is set to tomorrow, no matter what day of the week it is. 

So if a user generates a letter on a Friday, it would have Saturday's date on it. 
Things like Bank Holidays also weren't taken into account.

*currently affects 'Letter 1/2 in arrears...'
<img width="578" alt="image" src="https://user-images.githubusercontent.com/32230328/82997025-bf1b4880-9ffd-11ea-992a-ce0a4be94939.png">


### Changes in this PR
Now the preview generator will use the shiny new ✨ Bank Holidays Retriever ✨ to find the next working day to put on the letter. 

It uses the Government's [Bank Holidays API](https://www.gov.uk/bank-holidays.json) to get an array of dates - which is cached for a day to be reused for any other previews.

I had help from [this article](https://medium.com/just-tech/working-days-calculator-262e6637377f)! 🙏 

### How to test
- Pull the branch!
- Pick a Friday/day before bank holiday and use Timecop.freeze in `lib/hackney/pdf/preview_generator.rb`.
<img width="340" alt="image" src="https://user-images.githubusercontent.com/32230328/82997349-2cc77480-9ffe-11ea-92e5-176a5920fc97.png">
- `make run-all` in LBH-IncomeCollection
- Get any payment ref and send them a 'Letter 1/2 in arrears' and check the date there!

### Jira ticket 
https://hackney.atlassian.net/browse/MAAP-120?atlOrigin=eyJpIjoiZjU2ZTVhNGQ1Yjg0NGVjMmI5ZWVjY2VmZjY2OGU0MGUiLCJwIjoiaiJ9
